core/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/ShardsAllocator.java

```diff
@@ -52,5 +55,20 @@
      * @param shard shard to weigh
      * @return map of nodes to float weights
      */
+    @Deprecated
```

Comment:
just remove the API? In terms of BWC I think it's ok if we change the interface (people will have to recompile their plugins against 5.2 and they will see the change).